src/OAuthClient.php

```diff
@@ -19,217 +21,147 @@
-    protected $region;
-    /** @var string|null */
-    protected $grantCode;
-    /** @var string|null */
-    protected $redirectUri;
-    /** @var array<string> */
-    protected $scopes = ['AaaServer.profile.READ'];
-    /** @var bool */
-    protected $offlineMode = true;
-    /** @var bool */
-    protected $prompt = true;
-    /** @var string */
-    protected $state = 'test';
-    /** @var string */
-    protected $clientSecret;
-    /** @var string */
-    protected $clientId;
-    /** @var null|ZohoAccessToken */
-    protected $accessToken;
-    /** @var Cache\CacheItemPoolInterface|null */
-    protected $cache;
-    /** @var AbstractProvider */
-    protected $provider;
-    /** @var string */
-    protected $cachePrefix = 'zoho_crm_';
-    /** @var string|null */
-    protected $refreshToken;
-
-    public function __construct(string $clientId, string $clientSecret, string $region = Region::US, string $redirectUri = '')
-    {
-        $this->clientId = $clientId;
-        $this->clientSecret = $clientSecret;
-        $this->region = $region;
-        $this->redirectUri = $redirectUri;
+    protected ?string $grantCode = null;
+    protected string $state;
+    protected ?ZohoAccessToken $accessToken = null;
+    protected ?Cache\CacheItemPoolInterface $cache = null;
+    protected Zoho $provider;
+    protected string $cachePrefix = 'zoho_crm_';
+    protected ?string $refreshToken = null;
 
+    /**
+     * @param string $clientId
+     * @param string $clientSecret
+     * @param string $region
+     * @param string|null $redirectUri
+     * @param string[] $scopes
+     * @param bool $offlineMode
+     * @param bool $prompt
+     */
+    public function __construct(
+        protected string  $clientId,
+        protected string  $clientSecret,
+        protected string  $region = Region::US,
+        protected ?string $redirectUri = '',
+        protected array   $scopes = ['AaaServer.profile.READ'],
+        protected bool    $offlineMode = true,
+        protected bool    $prompt = true,
+    )
+    {
+        $this->state = uniqid('state_');
         $this->createProvider();
     }
 
+    /**
+     * @param string[] $additionalScopes
+     * @return string
+     */
     public function getAuthorizationUrl(array $additionalScopes = []): string
     {
         $scopes = array_unique(array_merge($this->scopes, $additionalScopes));
-
         $options = [
-            'scope' => $scopes,
-            'access_type' => $this->offlineMode ? 'offline' : 'online',
+            'scope'       => $scopes,
+            'access_type' => $this->offlineMode ? Mode::OFFLINE : Mode::ONLINE,
         ];
 
-        if($this->prompt){
+        if ($this->prompt) {
             $options['prompt'] = 'consent';
         }
 
         $url = $this->provider->getAuthorizationUrl($options);
-
         $this->state = $this->provider->getState();
 
         return $url;
     }
 
-    /**
-     * Get the oAuth2 state variable for security checks
```

Comment:
I think you should still keep the descriptions. They are helpful to know what the function does. 

Removing the return and params is fine because you've typed it anyways

Reply:
Hi! Thanks for the suggestion. I usually avoid comments that basically match the method name. Seem reduntant. Will try to keep the ones that make sense.
I'm currently waiting on this v5 to see if it makes senso to give it a complete overhaul instead of some small updates, but not sure i have the time.

Any suggestion for a refactoring / improvement are welcome!

Reply:
Ah yes that's how @xrobau works too. That makes sense. Maybe just keep it for stuff that needs an additional explanation. 

I like having it for IDE purposes. 

Everything else looked fine to me